slasher/detection/filter/filter.go

```diff
@@ -0,0 +1,94 @@
+package filter
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/spaolacci/murmur3"
+)
+
+// Filter is an encoded set of a []byte key.
+type Filter []byte
+
+// NewFilter returns a new Bloom filter that encodes the given key with 16 bits allotted for it.
+func NewFilter(key []byte) (Filter, error) {
+	nBits := 16
+	filter := make([]byte, 2)
+	for i := 0; i < len(filter); i++ {
+		filter[i] = 0
+	}
+
+	hash1 := hashutil.FastSum64(key)
```

Comment:
This feels like magic, can you add some comments to these parts?

Reply:
Added